dot/core/service.go

```diff
@@ -523,7 +526,19 @@ func (s *Service) GetMetadata(bhash *common.Hash) ([]byte, error) {
 	}
 
 	rt, err := s.blockState.GetRuntime(bhash)
-	if err != nil {
+	if errors.Is(err, blocktree.ErrFailedToGetRuntime) {
+		rt, err = s.getRuntimeFromDB(bhash)
+		if err != nil {
+			return nil, err
+		}
+		metadata, err := rt.Metadata()
+		if err != nil {
+			return nil, err
+		}
+		rt.Stop()
```

Comment:
move this one above in a defer statement. If `rt.Metadata()` fails, we still want to stop the runtime